### PR TITLE
Implement Doctor controller, dto, and services

### PR DESCRIPTION
**Pull request by ileanahi**
- Refactor Doctor entity to use doctorId instead of doctor
- Create doctor dto
- Remove Appointment repository from doctor repository

Create Services:
- Get all doctors
- Get doctor by Id
- Add doctor
- Update doctor
- Delete doctor

### CODE REVIEW

**Review by asya-code — APPROVED** [review]

**Review by VeronicaO27 — APPROVED** [review]
Changes look good to me

File: src/main/java/com/capstone/doctorsOffice/entities/Doctor.java [hunk 1]
```diff
@@ -77,5 +86,17 @@ public Set<Appointment> getAppointments() {
     public void setAppointments(Set<Appointment> appointments) {
         this.appointments = appointments;
     }
+
+    public Doctor(DoctorDto doctorDto) {
+        if (doctorDto.getId() != null) {
+            this.id = id;
+        }
```
Comment (williamsc51) [inline src/main/java/com/capstone/doctorsOffice/entities/Doctor.java:93]: Its the dto value that should be assigned here.
should be:
```
if (doctorDto.getId() != null) {
             this.id = doctorDto.getId();
         }
```

Change the other two below as well.

File: src/main/java/com/capstone/doctorsOffice/controllers/DoctorController.java [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+    @DeleteMapping("/{id}")
+    public void deleteDoctor(@PathVariable Long id) {
+        doctorService.deleteDoctor(id);
+    }
+
+    @PutMapping("/{id}")
+    public void updateDoctor(@PathVariable Long id, @RequestBody DoctorDto doctorDto) {
+        doctorService.updateDoctor(doctorDto);
+    }
```
Comment (williamsc51) [inline src/main/java/com/capstone/doctorsOffice/controllers/DoctorController.java:43]: You should pass the PathVariable `id` the the `updateDoctor` method in the service and use that id to look up the doctor.

```
doctorService.updateDoctor(id, doctorDto);
```

You would need to then update the function signature in `DoctorService` and `DoctorServiceImpl`

File: src/main/java/com/capstone/doctorsOffice/services/DoctorServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+    @Transactional
+    public void deleteDoctor(Long id) {
+        Optional<Doctor> doctorOptional = doctorRepository.findById(id);
+        doctorOptional.ifPresent(doctor -> doctorRepository.delete(doctor));
+    }
+
+    @Override
+    @Transactional
+    public void updateDoctor(DoctorDto doctorDto) {
+        Optional<Doctor> doctorOptional = doctorRepository.findById(doctorDto.getId());
```
Comment (williamsc51) [inline src/main/java/com/capstone/doctorsOffice/services/DoctorServiceImpl.java]: Use the id passed in from the controller to look up the Doctor. Reason: The Dto id could be null.
```
public void updateDoctor(Long id, DoctorDto doctorDto) {
         Optional<Doctor> doctorOptional = doctorRepository.findById(id);
```

**Review by williamsc51 — CHANGES_REQUESTED** [review]

**Review by williamsc51 — APPROVED** [review]

**Pull request merged**